project_euler/problem_081/sol1.py

```diff
@@ -0,0 +1,48 @@
+"""
+Problem 81: https://projecteuler.net/problem=81
+In the 5 by 5 matrix below, the minimal path sum from the top left to the bottom right,
+by only moving to the right and down, is indicated in bold red and is equal to 2427.
+
+    [131]   673   234    103    18
+    [201]  [96]  [342]   965   150
+     630   803   [746]  [422]  111
+     537   699   497    [121]  956
+     805   732   524    [37]  [331]
+
+Find the minimal path sum from the top left to the bottom right by only moving right
+and down in matrix.txt (https://projecteuler.net/project/resources/p081_matrix.txt),
+a 31K text file containing an 80 by 80 matrix.
+"""
+import os
+
+
+def solution(filename: str = "matrix.txt", n: int = 80) -> int:
+    """
+    Returns the minimal path sum from the top left to the bottom right of the matrix.
+    >>> solution()
+    427337
+    """
+    with open(os.path.join(os.path.dirname(__file__), filename), "r") as file:
+        data = file.read().strip().split("\n")
+
+    grid = []
+    for line in data:
+        row = line.split(",")
+        grid.append(list(map(int, row)))
```

Comment:
OPTIONAL:
```suggestion
def solution(filename: str = "matrix.txt") -> int:
    """
    Returns the minimal path sum from the top left to the bottom right of the matrix.
    >>> solution()
    427337
    """
    with open(os.path.join(os.path.dirname(__file__), filename), "r") as in_file:
        data = in_file.read()

    grid = [[int(cell) for cell in row.split(",")] for row in data.strip().splitlines()]
    dp = [[0 for cell in row] for row in grid]
    n = len(grid[0])
```
1. Remove `n` because we can calculate it by looking at the length of row 0 of the grid.
    * This the use of our algorithm and allows us to be flexible to handle other datasets.
    * This reduces the chances that the caller will give us bad data. (n = 79, n = 81, n = -5)
2. Used `in_file` instead of `file` to make it easier to find bugs.
    * `in_file.write()` would be a mistake because we can not write to a file opened in read mode.
    * `out_file.read()` would be a mistake because we can not read a file opened in write mode.
3. Do not strip() and split() the data right away because it can be useful to keep the raw data available.
4. Used a list comprehension inside a list comprehension to build the matrix.
    * As discussed in CONTRIBUTING.md, this repo (and Py3 in general) prefers comprehensions over `list(map())`